### PR TITLE
8345144: Robot does not specify all causes of  IllegalThreadStateException

### PR DESCRIPTION
**Pull request by alisenchung**
When robot.autoWaitForIdle is set to true, all mouse and key-related methods when invoked on the EDT will throw java.lang.IllegalThreadStateException which is not in the Robot specification.

This PR updates the specification by adding warnings to avoid calling lengthy and delay-type methods on EDT and including exceptions thrown when autoWaitForIdle is set to true and mouse/key-handling methods are called on the EDT.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Change requires CSR request [JDK-8346687](https://bugs.openjdk.org/browse/JDK-8346687) to be approved
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8345144](https://bugs.openjdk.org/browse/JDK-8345144): Robot does not specify all causes of  IllegalThreadStateException (**Bug** - P4)
 * [JDK-8346687](https://bugs.openjdk.org/browse/JDK-8346687): Robot does not specify all causes of  IllegalThreadStateException (**CSR**)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22564/head:pull/22564` \
`$ git checkout pull/22564`

Update a local copy of the PR: \
`$ git checkout pull/22564` \
`$ git pull https://git.openjdk.org/jdk.git pull/22564/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22564`

View PR using the GUI difftool: \
`$ git pr show -t 22564`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22564.diff">https://git.openjdk.org/jdk/pull/22564.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22564#issuecomment-2518701370)
</details>

### CODE REVIEW

**Comment by kumarabhi006** [comment]
Needs a CSR ?

**Review by honkar-jdk — COMMENTED** [review]
Doc changes look good to me. As @kumarabhi006 pointed, a corresponding CSR is required.

I have the same question as @azvegint suggested earlier: In case of robot.delay() is it good to have a recommendation - "to not call on EDT" or would it be better if the check is done by the method itself to avoid relying on the end-user to check it? 

```
public void delay(int ms) {
    if (isEventDispatchThread()) {
        throw new IllegalThreadStateException("delay() called on EDT");
    }
```

**Comment by azvegint** [comment]
> ```
> public void delay(int ms) {
>     if (isEventDispatchThread()) {
>         throw new IllegalThreadStateException("delay() called on EDT");
>     }
> ```

There is already a `checkNotDispatchThread` method, so we can reuse it.

File: src/java.desktop/share/classes/java/awt/Robot.java [hunk 1]
```diff
@@ -58,6 +58,14 @@
  * queue. For example, {@code Robot.mouseMove} will actually move
  * the mouse cursor instead of just generating mouse move events.
  * <p>
+ * Note: {@code waitForIdle()} must not be called on the AWT EDT, since
```
Comment (prrace) [inline src/java.desktop/share/classes/java/awt/Robot.java:61]: The word "and" needs to be before "since" for this to make sense and then that needs more rewording later.

Mulling it over, I suggest rewording like this

<pre>
 * Note: When {@code autoWaitForIdle()} is enabled, mouse and key related methods
 * cannot be called on the AWT EDT.
 * This is because when {@code autoWaitForIdle()} is enabled,
 * the mouse and key methods implicitly call {@code waitForIdle()}
 * which will throw {@code IllegalThreadStateException} when called on the AWT EDT.
</pre>

**Review by prrace — APPROVED** [review]
